internal/server/server.go

```diff
@@ -392,3 +396,30 @@ func configureEmail(options Options) {
 		email.SMTPServer = options.SMTPServer
 	}
 }
+
+func DefaultOptions(infraDir string) Options {
```

Comment:
What's the motivation for moving this to the server package?

Generally I would expect defaults to be set from the CLI, because defaults depend on the environment, and the CLI is what integrates the server package with a specific environment.

Reply:
Ah,I see
> it can set defaults for nested options without exposing internal structures like data.Options

Either way we have to export that struct, right? Why does it matter if it's important from `server` or from `cmd` (`cmd` has an indirect dependency on `data` either way).

Reply:
I think it's a common problem to have CLI configuration leak all over the app.

In a past role I had a colleague who has very insistent on handling all the configuration in the `cmd` or `main` package.  Everything would get constructed in one place, and the rest of the application didn't need to concern itself with configuration.

At first I was skeptical, but it worked out really well. 

I think our setup today is pretty close to that. If we were to move `server.New` into `cmd`, we could keep all of these `cmd` configuration options in one place. 

I think this might be moving us in the wrong direction, where configuration is now spread out everywhere instead of centralized in one place.

Reply:
Configuration is already spread out everywhere. When adding a new server option, you must add the field to `server.Options` then set defaults in `cmd.defaultServerOptions`. Not doing the latter can produce unexpected results.

> defaults depend on the environment, and the CLI is what integrates the server package with a specific environment

Maybe this is the main difference between our perspectives. I see the CLI as the entrypoint into the server package but ultimately the defaults are the server's defaults, not for the entrypoint. Adding another entrypoint doesn't necessarily change the defaults.

Reply:
The `server` package is a big risk. I've seen this pattern a few times now. It can very quickly become a large ball of integrating everything. It's the foundation of a difficult to decompose monolith.

Today our server is already:
1. the backend for API handlers
2. the package that contains API handlers, API routing, and API middlware
3. our IaaC implementation from config file
4. the management of goroutines for the `infra server` process

The changes here move even more of the `infra server` CLI command responsibilities into the `server` package, which I think is not really the right direction. We need to be very careful about expanding the responsibilities of `server`, and find opportunities to shrink it.

I think we should be looking to move the `server.New` function into `cmd` (or `cmd/server`) which would allow us to move `server.Options` into that package as well. The `server` package would receive a fully constructed `Server`. Accepting dependencies (instead of parameters used to construct dependencies) is often called dependency injection, and I have found it makes code much easier to test, and much easier to change.

By moving the defaults into `server`, I think we're moving away from a dependency injection approach, and doubling down on "passing in settings" instead of looking to pass in dependencies.

I think we could accomplish the goals of this PR without moving the defaults. In the short term I think the risk is small, so I don't want to block this PR, I'm ok if you merge it this way. But I also think it's likely we'll eventually be moving these defaults back into the `cmd` package.

Reply:
I want to avoid more tech debt so I'll update this PR without the default changes